src/sentry/event_manager.py

```diff
@@ -511,16 +512,13 @@ def save(self, project_id, raw=False, assume_normalized=False):
             id=project.organization_id
         )
 
-        # Check to make sure we're not about to do a bunch of work that's
-        # already been done if we've processed an event with this ID. (This
-        # isn't a perfect solution -- this doesn't handle ``EventMapping`` and
-        # there's a race condition between here and when the event is actually
-        # saved, but it's an improvement. See GH-7677.)
-        try:
-            event = Event.objects.get(project_id=project.id, event_id=data["event_id"])
-        except Event.DoesNotExist:
-            pass
-        else:
+        # Ensure an event with the same ID does not exist before processing it.
+        # We use a first write wins approach since Clickhouse cannot merge
+        # events from different days. (The timestamp rounded to
+        # start of day is part of the ordering key in Clickhouse).
```

Comment:
nit: primary key and not ordering key. Here they happen to be the same but in principle they may be different and it is the primary key that drives the merge.

Reply:
Thanks, updated